presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java

```diff
@@ -1211,4 +1213,9 @@ private static TypeSignature translateHiveUnsupportedTypeSignatureForTemporaryTa
             throw new UncheckedIOException(e);
         }
     }
+
+    public static Optional<String> getMetastoreHeader(ConnectorSession session)
+    {
+        return Optional.ofNullable(session.getProperty(METASTORE_HEADERS, String.class));
+    }
```

Comment:
I dont think you need this. We can directly call the getMetastoreHeaders() defined in HiveSessionProperties.

Reply:
✅